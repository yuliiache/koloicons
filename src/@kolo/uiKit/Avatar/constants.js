export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
