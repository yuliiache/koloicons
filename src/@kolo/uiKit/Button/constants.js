export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
