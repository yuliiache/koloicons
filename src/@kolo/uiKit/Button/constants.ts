export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
