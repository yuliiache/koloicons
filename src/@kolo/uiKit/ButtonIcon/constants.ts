export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
