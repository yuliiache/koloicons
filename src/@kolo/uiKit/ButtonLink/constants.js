export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

