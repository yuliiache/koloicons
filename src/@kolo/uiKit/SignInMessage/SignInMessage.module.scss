@import "@kolo/design";

.signInMessage {
  height: calc(100% - 64px);
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  padding: 1rem;
  font-size: 16px;
}

.signInText {
  line-height: 1.33;
  color: $lighthGrey;
  margin: 12px 0;
  text-align: center;
  font-size: 12px;
}

.signInTitle {
  &::first-letter {
    text-transform: capitalize;
  }
}
