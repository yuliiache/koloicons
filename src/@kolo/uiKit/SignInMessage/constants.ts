export const SIGN_IN_MESSAGE_TITLE_TEST_ID = 'signInMessageComponent';
