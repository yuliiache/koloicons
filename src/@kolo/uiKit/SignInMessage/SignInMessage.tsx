import {AppRoute} from 'constants/AppRoute';
import {FC} from 'react';

import {ButtonColor, ButtonSize} from '../Button/constants';
import ButtonLink from '../ButtonLink/ButtonLink';
import styles from '../SignInMessage/SignInMessage.module.scss';
import {SIGN_IN_MESSAGE_TITLE_TEST_ID} from './constants';

interface SignInMessageProps {
  title: string;
}

const SignInMessage: FC<SignInMessageProps> = ({title}) => {
  const URL = AppRoute.LOGIN;

  return (
    <div className={styles.signInMessage}>
      <p className={styles.signInText}>
        Sign In to work with
        <span
          className={styles.signInTitle}
          data-testid={SIGN_IN_MESSAGE_TITLE_TEST_ID}
        >
          {` ${title}`}
        </span>
      </p>
      <ButtonLink
        url={URL}
        text="Sign In"
        color={ButtonColor.BASIC}
        size={ButtonSize.FULL_WIDTH}
      />
    </div>
  );
};

export default SignInMessage;
