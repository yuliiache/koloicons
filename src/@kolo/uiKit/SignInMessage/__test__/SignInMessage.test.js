import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import {cleanup, render} from '@testing-library/react';

import {SIGN_IN_MESSAGE_TITLE_TEST_ID} from '../constants';
import SignInMessage from '../SignInMessage';

describe('SignInMessage', () => {
  afterEach(cleanup);

  const commonProps = {
    title: 'Download',
  };

  it('should render SignInMessage snapshot', () => {
    const {asFragment} = render(
      <Router>
        <SignInMessage {...commonProps} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render specific text based on title', () => {
    const {getByTestId} = render(
      <Router>
        <SignInMessage {...commonProps} />
      </Router>
    );

    const titleElement = getByTestId(SIGN_IN_MESSAGE_TITLE_TEST_ID);

    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent(`${commonProps.title}`);
  });
});
