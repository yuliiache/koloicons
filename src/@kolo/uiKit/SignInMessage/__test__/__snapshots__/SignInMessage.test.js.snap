// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInMessage should render SignInMessage snapshot 1`] = `
<DocumentFragment>
  <div
    class="signInMessage"
  >
    <p
      class="signInText"
    >
      Sign In to work with
      <span
        class="signInTitle"
        data-testid="signInMessageComponent"
      >
         Download
      </span>
    </p>
    <a
      data-testid="buttonLinkComponent"
      href="/login"
    >
      <button
        class="button button-full-width button-basic"
        data-testid="buttonComponent"
        type="button"
      >
        <span
          data-testid="buttonComponentText"
        >
          Sign In
        </span>
      </button>
    </a>
  </div>
</DocumentFragment>
`;
